### PR TITLE
Extended navigation examples (nesting navigation)

### PR DESCRIPTION
**Pull request by NickIliev**
An article that demonstrates a number of common scenarios for nesting forward and lateral navigations. Added schemes, description and playground demos fro TypeScript and Angular.

### CODE REVIEW

File: docs/core-concepts/navigation-extended.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            >>> Page
+```
+[Playground Demo TypeScript](https://play.nativescript.org/?template=play-tsc&id=LMV24L) 
+{% endnativescript %}
+
+{% angular %}
+![ng-nested-forward-in-forward-navigation](../img/navigation-extended/ng-navigation-examples-page-3.png?raw=true)
+
+
+Nesting a P-R-O inside a layout, for example, to create a secondary navigation level.
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
Nesting a P-R-O inside another page router outlet, for example, to create a secondary navigation level.
```

File: docs/core-concepts/navigation-extended.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+```
+
+[Playground Demo TypeScript](https://play.nativescript.org/?template=play-tsc&id=1UMjJZ)
+
+{% endnativescript %}
+
+{% angular %}
+![ng-navigation-lateral-in-forward](../img/navigation-extended/ng-navigation-examples-page-4.png?raw=true)
+
+Nesting a TabView (lateral navigation) inside a layout. For example in a Login component with forward navigation to a component with a nested TabView.
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
Nesting a TabView (lateral navigation) inside a page router outlet. For example in a Login component with forward navigation to a component with a nested TabView.
```

File: docs/core-concepts/navigation-extended.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+![navigation-schema-backward](../img/navigation-extended/navigation-examples-page-7.png?raw=true)
+
+Make an example where lots of these are combined. For example, RadSidedrawer + Login leading to a Page with TabView and in one TabView there is another TabView. In another, there might be a Frame that has secondary Frame navigation.
+
+[Playground Demo TypeScript](https://play.nativescript.org/?template=play-tsc&id=fyNqnr&v=6)
+RadSideDrawer > Frame 
+
+> **TODO** The Second TabView is not appearing on iOS
+> **TODO** Check if `tabTextFontSize` works for iOS
+> **TODO** iosOverflowSafeArea="false" for the nested TabView
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]: There are some `todos` here :)

File: docs/core-concepts/navigation-extended.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    tabItem > TabView (lateral navigation)
+                >> another content (e.g. layout or P-R-O)
+    tabItem > another content (e.g. layout or P-R-O)
+    tabItem > another content (e.g. layout or P-R-O)
+```
+[Playground Demo Angular](https://play.nativescript.org/?template=play-ng&id=ObeDAp)
+{% endangular %}
+
+
+ ## Multiple Navigation Scenarios
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
 ## Combining Nested Navigation Scenarios
```

**Review by MartoYankov — CHANGES_REQUESTED** [review]

**Comment by MartoYankov** [comment]
We should also revise the images.

File: docs/core-concepts/navigation-extended.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+{% nativescript %}
+![nested-forward-navigation](../img/navigation-extended/navigation-examples-page-1.png?raw=true)
+
+Nesting simple forward navigation: a `Frame` in a layout, for example, to show an advertisement banner on the top/bottom (static place).
+The root page is using a layout (e.g., a `GridLayout`) as a wrapper for the nested forward navigation (`Frame`) and for the static place (layout).
+```
+GridLayout  
+    > Frame (forward navigation)
+        >> Pages
+    > Another Layout
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
    > Static Content
```

File: docs/core-concepts/navigation-extended.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+{% angular %}
+![ng-nested-forward-navigation](../img/navigation-extended/ng-navigation-examples-page-1.png?raw=true)
+
+Nesting simple forward navigation: A `page-router-outlet` (mentioned in this article also as *P-R-O*) in a layout, for example, to show an advertisement banner on the top/bottom (static place).
+```
+GridLayout  
+    > P-R-O (forward navigation)
+        >> Components
+    > Another Layout (static place)
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
    > Static Content
```

File: docs/core-concepts/navigation-extended.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+{% nativescript %}
+![nested-lateral-navigation](../img/navigation-extended/navigation-examples-page-2.png?raw=true)
+
+Nesting simple lateral navigation: a TabView in a layout, for example, to show an advertisement banner on the top/bottom.
+```
+GridLayout  
+    > TabView (lateral navigation)
+        >> Pages | Layouts
+    > Another Layout (static place)
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
    > Static Content
```

File: docs/core-concepts/navigation-extended.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+{% angular %}
+![ng-nested-lateral-navigation](../img/navigation-extended/ng-navigation-examples-page-2.png?raw=true)
+
+Nesting simple lateral navigation: a TabView in a layout, for example, to show an advertisement banner on the top/bottom.
+```
+GridLayout  
+    > TabView (lateral navigation)
+        >> Layouts | P-R-Os
+    > Another Layout (static place)
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
    > Static Content
```

File: docs/core-concepts/navigation-extended.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+# Navigation Examples
+
+The main goal of the following article is to demonstrate some good practices for creating nested navigation structure. The article does not aim to be a strict guide, but will help you to understand how you could create complex navigation structures while using forward (e.g., frames or outlets) & lateral navigation (e.g., drawers, tab views, etc.). In each of the article sections, you can find visual guides along with corresponding Playground demos.
+
+## Nesting Simple Forward Navigation
+
+{% nativescript %}
+![nested-forward-navigation](../img/navigation-extended/navigation-examples-page-1.png?raw=true)
+
+Nesting simple forward navigation: a `Frame` in a layout, for example, to show an advertisement banner on the top/bottom (static place).
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
Nesting simple forward navigation: a `Frame` in a layout, for example, to show an advertisement banner on the top/bottom (static content).
```

File: docs/core-concepts/navigation-extended.md [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+The root page is using a layout (e.g., a `GridLayout`) as a wrapper for the nested forward navigation (`Frame`) and for the static place (layout).
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
The root page is using a layout (e.g., a `GridLayout`) as a wrapper for the nested forward navigation (`Frame`) and for the static content (layout).
```

File: docs/core-concepts/navigation-extended.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+    > Another Layout
+```
+
+Code: [Playground Demo TypeScript](https://play.nativescript.org/?template=play-tsc&id=65Uk0F)
+{% endnativescript %}
+
+{% angular %}
+![ng-nested-forward-navigation](../img/navigation-extended/ng-navigation-examples-page-1.png?raw=true)
+
+Nesting simple forward navigation: A `page-router-outlet` (mentioned in this article also as *P-R-O*) in a layout, for example, to show an advertisement banner on the top/bottom (static place).
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
Nesting simple forward navigation: A `page-router-outlet` (mentioned in this article also as *P-R-O*) in a layout, for example, to show an advertisement banner on the top/bottom (static content).
```

File: docs/core-concepts/navigation-extended.md [hunk 12]
```diff
@@ -0,0 +1,219 @@
+---
+title: Navigation (extended examples)
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
title: Nested Navigation
```

File: docs/core-concepts/navigation-extended.md [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+description: Learn how to configure your application navigation architecture, navigate forward and backward and use TabView, Modal View and SideDrawer
+position: 44
+slug: extended-navigation
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
slug: nested-navigation
```

File: docs/core-concepts/navigation-extended.md [hunk 14]
```diff
@@ -0,0 +1,219 @@
+---
+title: Navigation (extended examples)
+description: Learn how to configure your application navigation architecture, navigate forward and backward and use TabView, Modal View and SideDrawer
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
description: Learn how to nest navigation widgets and implement complex navigation scenarios
```

File: docs/core-concepts/navigation-extended.md [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+position: 44
+slug: extended-navigation
+environment: nativescript
+---
+
+# Navigation Examples
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
# Nested Navigation
```

**Comment by tsonevn** [comment]
test

File: docs/core-concepts/navigation-extended.md [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+
+The main goal of the following article is to demonstrate some good practices for creating nested navigation structure. The article does not aim to be a strict guide, but will help you to understand how you could create complex navigation structures while using forward (e.g., frames or outlets) & lateral navigation (e.g., drawers, tab views, etc.). In each of the article sections, you can find visual guides along with corresponding Playground demos.
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
The main goal of the following article is to demonstrate some good practices for creating nested navigation structure. The article does not aim to be a strict guide, but will help you to understand how you could create complex navigation structures while using forward (e.g., frames or outlets) & lateral navigation (e.g., drawers, tab views, etc.). In each of the article sections, you can find visual guides along with corresponding Playground demos.

#Simple Rule

There is one simple rule when it comes to nesting navigation widgets.

{% nativescript %}
> **Important:** When nesting a frame or a tab-view, they should never have direct siblings in the markup. Instead, wrap them in a layout and nest this layout.
{% endnativescript %}
{% angular %}
> **Important:** When nesting an outlet or a tab-view, they should never have direct siblings in the markup. Instead, wrap them in a layout and nest this layout.
{% endangular %}

If these components have siblings, they will span over them in most scenarios. The reason for this is on iOS the navigation controllers always take all the space provided by their parent regardless of their own layout parameters.

You can check out how this is done in the examples below.
```

File: docs/core-concepts/navigation-extended.md [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+```
+[Playground Demo Angular](https://play.nativescript.org/?template=play-ng&id=hBdlPB)
+{% endangular %}
+
+## Nesting Forward in Forward Navigation
+
+{% nativescript %}
+![nested-forward-in-forward-navigation](../img/navigation-extended/navigation-examples-page-3.png?raw=true)
+
+Nesting a Frame inside a Page/Frame, for example, a secondary navigation level.
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
Nesting a Frame inside a Page/Frame, for example, a secondary navigation level.

> **Note:** Each `Frame` comes with its own `ActionBar` by default. It's typical that you want to keep one `ActionBar` on top of the screen when nesting navigations. Set the `actionBarVisibility` property of the `Frame` to `never` to hide the `ActionBar` where needed.
```

File: docs/core-concepts/navigation-extended.md [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+            >>> Page
+```
+[Playground Demo TypeScript](https://play.nativescript.org/?template=play-tsc&id=LMV24L) 
+{% endnativescript %}
+
+{% angular %}
+![ng-nested-forward-in-forward-navigation](../img/navigation-extended/ng-navigation-examples-page-3.png?raw=true)
+
+
+Nesting a page router outlet inside another page router outlet, for example, to create a secondary navigation level.
```
Comment (MartoYankov) [inline docs/core-concepts/navigation-extended.md]:
```suggestion
Nesting a page router outlet inside another page router outlet, for example, to create a secondary navigation level.

> **Note:** Each page router outlet comes with its own `ActionBar` by default. It's typical that you want to keep one `ActionBar` on top of the screen when nesting outlets. Set the `actionBarVisibility` property of the page router outlet to `never` to hide the `ActionBar` where needed.
```

**Review by MartoYankov — CHANGES_REQUESTED** [review]

**Pull request merged**